src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -0,0 +1,238 @@
+import "@polymer/paper-tooltip/paper-tooltip";
+import "@material/mwc-button";
+import "@polymer/iron-icon/iron-icon";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-item/paper-item-body";
+
+import "../../../components/ha-card";
+import "../../../components/ha-data-table";
+import "../../../components/entity/ha-state-icon";
+import "../../../layouts/hass-subpage";
+import "../../../resources/ha-style";
+import "../../../components/ha-icon-next";
+
+import "../ha-config-section";
+// import { computeRTL } from "../../../common/util/compute_rtl";
+
+import memoizeOne from "memoize-one";
+
+import {
+  LitElement,
+  html,
+  TemplateResult,
+  property,
+  customElement,
+} from "lit-element";
+import { HomeAssistant } from "../../../types";
+// tslint:disable-next-line
+import {
+  DataTabelColumnContainer,
+  RowClickedEvent,
+  DataTabelRowData,
+} from "../../../components/ha-data-table";
+// tslint:disable-next-line
+import { DeviceRegistryEntry } from "../../../data/device_registry";
+import { EntityRegistryEntry } from "../../../data/entity_registry";
+import { ConfigEntry } from "../../../data/config_entries";
+import { AreaRegistryEntry } from "../../../data/area_registry";
+import { navigate } from "../../../common/navigate";
+
+interface DeviceRowData extends DeviceRegistryEntry {
+  device?: DeviceRowData;
+  area?: string;
+  integration?: string;
+  battery_entity?: string;
+}
+
+@customElement("ha-config-devices-dashboard")
+export class HaConfigDeviceDashboard extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public narrow = false;
+  @property() public devices!: DeviceRegistryEntry[];
+  @property() public entries!: ConfigEntry[];
+  @property() public entities!: EntityRegistryEntry[];
+  @property() public areas!: AreaRegistryEntry[];
+  @property() public domain!: string;
+
+  private _devices = memoizeOne(
+    (
+      devices: DeviceRegistryEntry[],
+      entries: ConfigEntry[],
+      entities: EntityRegistryEntry[],
+      areas: AreaRegistryEntry[],
+      domain: string
+    ) => {
+      let outputDevices: DeviceRowData[] = [...devices];
+      if (domain) {
+        outputDevices = outputDevices.filter(
+          (device) =>
+            entries.find((entry) =>
+              device.config_entries.includes(entry.entry_id)
+            )!.domain === domain
+        );
+      }
+
+      outputDevices = outputDevices.map((device) => {
+        device.name = device.name_by_user || device.name;
```

Comment:
You are mutatin data that you do not own. You should make a copy.

Reply:
If you do `[...devices]`, you're only copying the array, the objects are not copied. (shallow copy)

Reply:
Should we have a "<no name>" placeholder ?

Reply:
I even wonder if we should extract that into a helper, we had the same problem in the device picker. Should also be localized.

Reply:
Helper is a good idea I think, let's add that in a separate PR